### PR TITLE
Add scrollToIndex to GroupedList, DetailsList

### PR DESCRIPTION
**Pull request by elliottsj**
#### Pull request checklist

- [ ] Addresses an existing issue
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Add the `scrollToIndex` method to DetailsList and GroupedList.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx [hunk 1]
```diff
@@ -120,6 +120,11 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
     this._initialFocusedIndex = props.initialFocusedIndex;
   }
 
+  public scrollToIndex(index: number, measureItem?: (itemIndex: number) => number): void {
+    this._list && this._list.scrollToIndex(index, measureItem);
+    this._groupedList && this._groupedList.scrollToIndex(index, measureItem);
+  }
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx]: Can't we have a `componentRef` for `IList` on `DetailsList`? `listRef`?

Comment (elliottsj) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx]: Not sure what you mean by `componentRef`. Can you provide an example?

Comment (cschleiden) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx]: every component has a `componentRef` property that gives you the instance of the component. if you use plain `List` directly and you use `componentRef` property, you'll get an instance of `IList`. We could use a similar property to return `this._list || this._groupedList` to the user of `DetailsList`.. but now that I think more about it, what we have here might be fine, the underlying list is relatively transparent to the user of `DetailsList` and maybe we shouldn't leak that

**Comment by micahgodbolt** [comment]
fixes #2757

**Pull request merged**